internal/state/ownership/tree.go

```diff
@@ -0,0 +1,180 @@
+type tree struct {
+	contract common.Address
+	mt       merkletree.MerkleTree
+	store    storage.Tx
+	tx       bool
+}
+
+// NewEnumeratedTokenTree creates a new merkleTree with a custom storage
+func NewTree(contract common.Address, store storage.Tx) (Tree, error) {
+	if contract.Cmp(common.Address{}) == 0 {
+		return nil, fmt.Errorf("contact address is " + common.Address{}.String())
+	}
+
+	t, err := sparsemt.New(TreeDepth, store, contract.String()+TreePrefix)
+	if err != nil {
+		return nil, err
+	}
+
+	root, err := headRoot(contract, store)
+	if err != nil {
+		return nil, err
+	}
+
+	t.SetRoot(root)
+	slog.Debug("enumeratedTokenTree", "HEAD", root.String())
+
+	return &tree{contract, t, store, false}, err
+}
+
+// TokenTransfer updates the SlotOwner of the token
+func (b *tree) Transfer(eventTransfer scan.EventTransfer) error {
+
+	tokenData, err := b.TokenData(eventTransfer.TokenId)
+	if err != nil {
+		return err
+	}
+
+	tokenData.SlotOwner = eventTransfer.To
+	if err := b.SetTokenData(tokenData, eventTransfer.TokenId); err != nil {
+		return err
+	}
+	return setHeadRoot(b.contract, b.store, b.Root())
+}
+
+// Mint creates a new token
+func (b *tree) Mint(tokenId *big.Int, idx int) error {
+	tokenData, err := b.TokenData(tokenId)
+	if err != nil {
+		return err
+	}
+
+	tokenData.Minted = true
+	tokenData.Idx = idx
+	if err := b.SetTokenData(tokenData, tokenId); err != nil {
+		return err
+	}
+
+	return setHeadRoot(b.contract, b.store, b.Root())
+}
+
+// SetTokenData updates the tokenData
+func (b *tree) SetTokenData(tokenData *TokenData, tokenId *big.Int) error {
+	buf, err := json.Marshal(tokenData)
+	if err != nil {
+		return err
+	}
+
+	hash := crypto.Keccak256Hash(buf)
+	if err := b.store.Set([]byte(b.contract.String()+TokenDataPrefix+hash.String()), buf); err != nil {
+		return err
+	}
+
+	return b.mt.SetLeaf(tokenId, hash)
+}
+
+// TokenData returns the tokenData
+func (b *tree) TokenData(tokenId *big.Int) (*TokenData, error) {
+	leaf, err := b.mt.Leaf(tokenId)
+	if err != nil {
+		return &TokenData{common.Address{}, false, 0}, err
+	}
+	if leaf.String() == sparsemt.Null {
+		SlotOwner := initSlotOwner(tokenId)
+		return &TokenData{SlotOwner, false, 0}, nil
+	}
+
+	buf, err := b.store.Get([]byte(b.contract.String() + TokenDataPrefix + leaf.String()))
+	if err != nil {
+		return &TokenData{common.Address{}, false, 0}, err
+	}
+
+	var tokenData TokenData
+	if err := json.Unmarshal(buf, &tokenData); err != nil {
+		return &TokenData{common.Address{}, false, 0}, err
+	}
+
+	return &tokenData, nil
+}
+
+func (b *tree) OwnerOf(tokenId *big.Int) (common.Address, error) {
+	tokenData, err := b.TokenData(tokenId)
+	if err != nil {
+		return common.Address{}, err
+	}
+
+	if tokenData.Minted {
+		return tokenData.SlotOwner, nil
+	}
+
+	return common.Address{}, nil
+}
+
+func initSlotOwner(tokenId *big.Int) common.Address {
+	SlotOwner, _ := new(big.Int).SetString(Ones160bits, 16) // omit success. this is constant and it will always be true
```

Comment:
Make `SlotOwner` unexported, please

Reply:
I will change this to lowercase. But, I am exporting SlotOwner Minted, and Idx because other trees will need.